### ISSUE TITLE
[5.x]: Matrix propagation sometimes wrong for new entry

### ISSUE THREAD

**Issue by WHITE-developer** [craftcms/cms#15321]
### Description
In our large multi-site multi-site-group setup it seems that some sites are getting the matrix blocks propagated while they shouldn't (as of Craft 5, Craft 4 propagated always on first save?).
The matrix is set to "Save entries to other sites in the same site group" but somehow sites from another group also get the blocks propagated, sometimes even only one site from another site group.

### Steps to reproduce

1. Have a multi-site multi-site-group setup (in our case there are 58)
2. Have a matrix set to "Save entries to other sites in the same site group" 
3. Create an entry with a matrix block in it
4. Save it

### Expected behavior
All sites should have the "first time save" matrix content.
But at least only sites in the same site group have the matrix field.

### Actual behavior
Some random sites from other groups also have the matrix field propagated.

(Same project as with the validation visibility, so should be reproducible with the DB sent last time)

### Craft CMS version

5.2.5

### PHP version

8.3.7

### Database type and version

MySQL 8.0.37

**Comment by i-just** [comment]
Hi, thanks for getting in touch! Can you please provide more info on how to replicate? The matrix field is set to the “Save entries to other sites in the same site group” propagation method. What’s the propagation method of the section to which it is added? I found the database export you sent over before. Can you please point to which entry and field are exhibiting this problem?

**Comment by WHITE-developer** [comment]
Hi @i-just 
I did answer Brandon in that email conversation on this last Wednesday 09:51 CEST.
I discovered it in the news section. Section is set to "Save entries to all sites enabled for this section".
Happens with the "contentBlocks" matrix. It propagated to for example "Turkey" and "Slovakia", while those are other groups then when you start in "International". "Czech Republic" is in the same group as "Slovakia", but that one didn't get propagated.
